homeassistant/components/aemet/sensor.py

```diff
@@ -25,6 +35,13 @@
 )
 from .weather_update_coordinator import WeatherUpdateCoordinator
 
+RENAMED_KEYS = {
+    ATTR_FORECAST_NATIVE_PRECIPITATION: ATTR_FORECAST_PRECIPITATION,
+    ATTR_FORECAST_NATIVE_TEMP: ATTR_FORECAST_TEMP,
+    ATTR_FORECAST_NATIVE_TEMP_LOW: ATTR_FORECAST_TEMP_LOW,
+    ATTR_FORECAST_NATIVE_WIND_SPEED: ATTR_FORECAST_WIND_SPEED,
+}
```

Comment:
So the entity descriptions are still renamed, instead of keeping this backward compatible thing around forever, it might be easier/better to migrate existing entries in that case.

E.g.:  <https://github.com/home-assistant/core/blob/dev/homeassistant/components/plugwise/gateway.py#L31>

Since we know the keys before and after, that would be a more solid solution.

Reply:
Upgrade to 2022.7 followed by a downgrade to a previous release won't be backwards compatible if entity's unique ID is changed.
Is it better to be backwards compatible for 2-3 releases and then add the migration?
Or we can handle this in aemet's weather entity instead?

Reply:
> Upgrade to 2022.7 followed by a downgrade to a previous release won't be backwards compatible if entity's unique ID is changed.

We don't handle downgrades on registry/config entries. We never did.
Downgrades are available via backup.